### DISCUSSION TITLE
Keep modified config files after upgrade

### DISCUSSION THREAD

**Discussion by Sahale** [Security-Onion-Solutions/securityonion#4700] [Ideas] [answered]
Hello!
I've modified "/opt/so/saltstack/default/salt/logstash/pipelines/config/so/0900_input_redis.conf.jinja" and  "/opt/so/saltstack/default/salt/logstash/pipelines/config/so/9999_output_redis.conf.jinja" for my purposes.
After upgrade from 2.3.52 to 2.3.60 modified files was overwritten.
Is it possible to make an option to keep some modified configs after upgrade? Thanks!

**Comment by weslambert** [comment, marked as answer]
You should be able to copy files from the directories in `/opt/so/salstack/default/` to the equivalent directory under `/opt/so/saltstack/local/` and the changes should be preserved. Keep in mind, you might need to update these files during upgrades.  We have an issue open to identify when local changes are present during upgrades here: https://github.com/Security-Onion-Solutions/securityonion/issues/3860

**Reply by facyber** [reply]
Hey @weslambert 

Just to be clear is it recommended to move files to the local directory or to copy them? So far I practiced to copy them to the equivalent directory, and even after upgrades haven't noticed issues so far.

**Reply by weslambert** [reply]
I've adjusted my comment to be clearer -- you should copy them.  While you may not have experienced any issues yet, we've had several folks experience issues with incompatible configuration settings for downstream applications, etc, so I would keep that in mind and be aware of the potential.  The issue referenced above should help with that awareness.

**Reply by facyber** [reply]
I will then review my files that are changed so far. Good to know and thanks for clarification.
